---
title: "analysis"
author: "M. Vicari marco.vicari@scilifelab.se"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document:
    theme: cosmo
    highlight: tango
    css: style.css
    code_folding: "hide"
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: false
    number_sections: false
editor_options: 
  chunk_output_type: console
---

In this script we will perform all the calculations and analysis needed to show the results presented in the paper. The R objects and the tables needed to plot the results will be saved to be used in the main.Rmd file. 

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)

```

Set-upLoad libraries
```{r message=FALSE, warning=FALSE}

library(dplyr)
library(STutility)
library(stringr)
library(RColorBrewer)
library(harmony)
library(tidyr)
library(ggrepel)
# library(data.table)
# library(magrittr)
# library(knitr)

```

Set-up directories and system variables
```{r message=FALSE, warning=FALSE}

project.dir <- ".."
data.dir <- paste0(project.dir, "/data")
dir.create(paste0(project.dir, "/results"))
dir.create(paste0(project.dir, "/results/plots"))
plots.dir <- paste0(project.dir, "/results/plots")
tables.dir <- paste0(project.dir, "/results/tables")

my_cols = brewer.pal(12, "Paired")
```

# 1 Correlation analysis
For this analysis we will use only Striatum data.

## Upload striatum object
First of all we assemble spaceranger output files and load them into a `Seurat` object. For this analysis we will use sections from the Striatum of 3 different 6-OHDA mice (mPD1, mPD3 and mPD4), so that we are sure that they come from the same biological condition. To check the quality of RNA data produced with sma using different matrices and compared them with Visium internal controls (iCtrl) or standalone Visium.
```{r}
se <- readRDS(file = paste0(project.dir, "/R_objects/mPDStrRNA.10xFFBM.STUnF"))
```


## Colors by type
```{r}

type.area.bymatrix.cols <- c(rep(NA, length(levels(as.factor(se$type)))))
names(type.area.bymatrix.cols) <- levels(as.factor(se$type))
type.area.bymatrix.cols[grepl("DHB", names(type.area.bymatrix.cols))] <- "darkolivegreen3"
type.area.bymatrix.cols[grepl("FMP10", names(type.area.bymatrix.cols))] <- "purple"
type.area.bymatrix.cols[grepl("9AA", names(type.area.bymatrix.cols))] <- "gold"
type.area.bymatrix.cols[grepl("nM", names(type.area.bymatrix.cols))] <- "lightgrey"

```

## Calculate gene attributes (Log umi count per gene)
Here I calculate Log10UMI count for every gene in every condition. Then I plot scatter plot of every technical or biological condition pairs, and calculate pearson correlation between them.

```{r warning=FALSE, message=FALSE}

gene_attr <- do.call(rbind, lapply(unique(se$type), function(i) {
  umis <- GetAssayData(se, slot = "counts", assay = "RNA")[, se$type %in% i]
  gene_attr <- data.frame(gene = rownames(umis),
                        counts = Matrix::rowSums(umis), 
                        det_rate = Matrix::rowMeans(umis > 0),
                        dataset = i)
}))

gene_attr$log10_count <- log10(gene_attr$counts + 1)
write.csv(gene_attr, file = paste0(project.dir, "/results/tables/gene_attr_RNA.mPDStr.10xF.STUnF.csv"))

Log10countsmPDStr <- spread(data = gene_attr[, c(1, 4, 5)], key = "dataset", value = "log10_count")
write.csv(Log10countsmPDStr, file = paste0(project.dir, "/results/tables/Log10counts_RNA.mPDStr.10xF.STUnF.csv"))

Log10countsmPD3 <- Log10countsmPDStr[ , grepl( "mPD3" , names( Log10countsmPDStr ) ) ]
write.csv(Log10countsmPD3, file = paste0(project.dir, "/results/tables/Log10countsmPD3_RNA.mPDStr.10xF.STUnF.csv"))

Log10countsFMP10 <- Log10countsmPDStr[ , grepl( "FMP10|M.mPD" , names( Log10countsmPDStr ) ) ]
write.csv(Log10countsFMP10, file = paste0(project.dir, "/results/tables/Log10countsFMP10_RNA.mPDStr.10xF.STUnF.csv"))

```

## Calculate correlations and their significance. 
Here I calculate the correlations again with another function just to have them stored in a table, so I can plot just the correlation coefficients. I also calculate the significance of each correlation that could be used in a different version of the plot (not done here). 

```{r warning=FALSE, message=FALSE}

gene_attr_wide <- Log10countsmPDStr
rownames(gene_attr_wide) <- gene_attr_wide$gene
gene_attr_wide$gene <- NULL
gawmPD3 <- gene_attr_wide[ , grepl( "mPD3" , names( gene_attr_wide ) ) ]
gawFMP10 <- gene_attr_wide[ , grepl("FMP10|M.mPD", names( gene_attr_wide ) ) ]

corrm <- cor(gene_attr_wide)
write.csv(corrm, file = paste0(project.dir, "/results/tables/corrmLog10counts_RNA.mPDStr.10xF.STUnF.csv"))
corrmPD3 <- cor(gawmPD3)
write.csv(corrmPD3, file = paste0(project.dir, "/results/tables/corrmPD3Log10counts_RNA.mPDStr.10xF.STUnF.csv"))
corrFMP10 <- cor(gawFMP10)
write.csv(corrFMP10, file = paste0(project.dir, "/results/tables/corrFMP10Log10counts_RNA.mPDStr.10xF.STUnF.csv"))

cor.mtest <- function(mat, ...) {
    mat <- as.matrix(mat)
    n <- ncol(mat)
    p.mat<- matrix(NA, n, n)
    diag(p.mat) <- 0
    for (i in 1:(n - 1)) {
        for (j in (i + 1):n) {
            tmp <- cor.test(mat[, i], mat[, j], ...)
            p.mat[i, j] <- p.mat[j, i] <- tmp$p.value
        }
    }
  colnames(p.mat) <- rownames(p.mat) <- colnames(mat)
  p.mat
}

p.matm <- cor.mtest(gene_attr_wide)
write.csv(p.matm, file = paste0(project.dir, "/results/tables/p.matmLog10counts_RNA.mPDStr.10xF.STUnF.csv"))
p.matmPD3 <- cor.mtest(gawmPD3)
write.csv(p.matmPD3, file = paste0(project.dir, "/results/tables/p.matmPD3Log10counts_RNA.mPDStr.10xF.STUnF.csv"))
p.matFMP10 <- cor.mtest(gawFMP10)
write.csv(p.matFMP10, file = paste0(project.dir, "/results/tables/p.matFMP10Log10counts_RNA.mPDStr.10xF.STUnF.csv"))

```

# 2 Gene body coverage

```{r}

V11L12_109_D1_pgb <- c(0.0,0.0022391332907578504,0.003502076154854637,0.0033208556551917153,0.003880919009558361,0.0042612020271733135,0.004719093823893359,0.004482827097401258,0.004950720025449279,0.006686276351580891,0.008981255959674158,0.011170223575690882,0.012340835995367797,0.013915494120694934,0.014809515252365348,0.016370891875288934,0.01773448612507047,0.017289395776450233,0.016966799284335043,0.018817728661466186,0.01859482344642826,0.019879768799446595,0.0195894159547328,0.019515087546703284,0.020872841135789282,0.02244549903485082,0.023706441672682012,0.02437667748975764,0.0257521130790315,0.026967290539905875,0.028057973918009606,0.030193335470108378,0.03240718590086911,0.033494028844542896,0.0351798195411865,0.03850827605618748,0.04043289376894314,0.04280780241860959,0.046626954441086374,0.049796512981548476,0.052981513268780964,0.05458209432651021,0.05622420008151322,0.05983036801122943,0.06301456820795569,0.06423974680015804,0.06885490886729106,0.07565967862284581,0.07887396222260662,0.0826164655745858,0.08734436039404778,0.09218050745900376,0.09665645377805138,0.10033599000718961,0.1048703429331926,0.11048993862185691,0.11579669892258213,0.11944151122374963,0.1253698618387712,0.1283683610289996,0.13412725247479995,0.13916670253739102,0.14360848499182227,0.14819188346585765,0.15303155092904108,0.1578388947357725,0.16026452912353445,0.16696224676935456,0.17335849031242415,0.18152973464358288,0.18873006914542173,0.1945703297957065,0.20184995326671354,0.2096865997481635,0.22220177546484057,0.23105605706117477,0.24248422981607679,0.25683737344368396,0.27081407448810607,0.28420150887468393,0.2978982582509733,0.3151739724797669,0.3344247901322598,0.3575878103331049,0.3808096371861585,0.4103408177533043,0.4460772602596806,0.4816274416962047,0.5233215181301308,0.5712013342949355,0.6213494070449249,0.6803573620247922,0.7451751742157072,0.8262983508694504,0.9125941926550731,0.9795812102265008,1.0,0.9563581032286372,0.7559355114250524,0.37524300748900713)
V11L12_109_C1_RNA_pgb <- c(0.0,0.003023274954878687,0.005244518851460773,0.005325807310851139,0.006235921494568032,0.0068927865141202235,0.007272094972054407,0.007482721397428888,0.00741545208819764,0.009340711024162231,0.011778234229247475,0.013705415145475817,0.01467228427580462,0.016339771741101572,0.017102571790485157,0.018799680363376373,0.01978362120075378,0.0190681923119885,0.018347449713082265,0.02022884227596833,0.019135687736544893,0.020686725809391113,0.020491022995476338,0.01972403981257753,0.02127134698255882,0.022952175252039447,0.02447008742975162,0.024864093383820365,0.026154533544434602,0.02735147501810053,0.02803061239717971,0.02996457677316246,0.03267728232895167,0.033617017620262826,0.03551977808137529,0.03912609140214565,0.04140194214975075,0.044086044116908855,0.047623392263509375,0.05108985325567628,0.054060669455120856,0.05633414599181191,0.05721656104819829,0.061507212400526215,0.06500691234548975,0.06534902483243724,0.06958506933374214,0.07775303322402918,0.08111435059000272,0.0848209460574758,0.08989440866546283,0.0948241749843189,0.09969300322304785,0.10352090956246232,0.10854553126021758,0.11494504719252951,0.12073868711110991,0.1257806066312389,0.13125553699902454,0.13413206310788278,0.14074842363701073,0.14638694844254024,0.14993944631592562,0.15521528213991925,0.16087958409251554,0.16659894512678514,0.16968643683400558,0.17790674642206417,0.1837847273569244,0.19117869848923308,0.1989619271537824,0.20470028187536432,0.21281578701021725,0.22098770791869435,0.2326401089694975,0.23979812876001522,0.2500798187097769,0.26399054657048626,0.277721173574716,0.29082840062751525,0.30335632021728776,0.31899569973874636,0.3370600530647445,0.35777493023211643,0.3780324891543784,0.40322965041214043,0.4350222565314542,0.46840140082966236,0.509199728118933,0.5571503092579302,0.6090722669101476,0.6683913613092258,0.7315280738004243,0.8068235726130927,0.8926327556991237,0.9698098731900033,1.0,0.9643924891724677,0.7430329346015689,0.3511119899460082)
V11L12_038_A1_RNA_pgb <- c(0.0,0.005141834665768967,0.009152949981211394,0.009535397248447666,0.01198321682129255,0.013604102159230257,0.014648306231102595,0.01498638333283815,0.016258458962636002,0.019585305177082117,0.02345074499340272,0.02543968019974166,0.025364551954911534,0.02730820079764207,0.028095214972142995,0.029644505972237365,0.030318304237717284,0.02924739953527814,0.027092370700560413,0.029233394792774967,0.02835161955685068,0.02819272462788941,0.029178553791757857,0.029239022866865028,0.031924792176819955,0.03445978145533881,0.036259848966050806,0.036733916044055515,0.03779186308754312,0.03929246470248626,0.03998759729533083,0.04246460432217766,0.045903226705761274,0.04713407342071333,0.04987782498234028,0.054028202410150744,0.056033367411540094,0.058788636892235076,0.06176680428398529,0.0653238779943481,0.06860059508375949,0.07134604815615785,0.07399817990701636,0.07933856779126494,0.08354430976186833,0.08570391958712449,0.09042168541447952,0.098569827842449,0.10304571736937534,0.10677765403286194,0.11149489631844114,0.1162866124216307,0.12179911466468,0.12681058742826987,0.1314821506939088,0.13768494276837634,0.14312310207925924,0.14756221279643425,0.15386264541209216,0.15666006000574065,0.1633659757267709,0.16899575132760378,0.17328931743110093,0.17894108178651824,0.1849310541291151,0.1914901163820279,0.19572949591172772,0.2036595831900683,0.21012388438155777,0.21778631092701056,0.22432456239433454,0.23004805197303918,0.23756650453014155,0.24412739917926973,0.2565641340638661,0.2637823354125359,0.2744678230570154,0.2875296668219404,0.2989334537828446,0.30942627805382245,0.32082378251341676,0.3365165545872796,0.3527442554705862,0.37244958259977495,0.3946214459202942,0.423364543840014,0.4567104901672965,0.49574026778376523,0.537809598065199,0.5850489033828521,0.6360435738584597,0.6985611370489704,0.7663672574879236,0.8464571077274897,0.927946254248378,0.9854742025444392,1.0,0.9223561369370633,0.660836810850508,0.29691886502949044)
V11L12_109_A1_RNA_pgb <- c(0.0,0.002967063921792624,0.0045434325104864995,0.0046222310747369285,0.006276073882063579,0.006193434715521953,0.00644400090638214,0.005803414931122771,0.0061721127510777195,0.008315698567812918,0.01146817087607714,0.013905231681684525,0.015545831032834634,0.01728721308548823,0.01826934779554064,0.02034300816366369,0.02142354175310781,0.020345656854899,0.018729028159428434,0.02056404144725019,0.01966070530144821,0.02074997957196885,0.02121151401972136,0.020966377645893555,0.023023616128357696,0.025078338354148293,0.02670450233806597,0.02728575762965443,0.028995355387484326,0.030483787427165957,0.03121522351179641,0.03313194892422743,0.036409439458798216,0.03779073193801162,0.0402565310435221,0.04418348066899054,0.046489431258450155,0.049330947215689154,0.05418507120807734,0.05776398280522624,0.06145692056005517,0.06330014479070638,0.06499914778359504,0.06956681581888462,0.07357468296159173,0.07521130927588887,0.08080561003398404,0.0893260524342863,0.09295343508104134,0.09737568996751247,0.10316480196596459,0.10906701308016437,0.11488605528957545,0.11928592873510822,0.12439008918010955,0.13191780210541817,0.1383844493218622,0.14336624023179226,0.15050181441971347,0.15373070147011633,0.16098771815117643,0.16677908153717855,0.17094097007521886,0.17665472680802644,0.18269347795540636,0.1888670474866635,0.19218373865151686,0.20051294554462984,0.206961449226112,0.2149171906550465,0.22281823660997188,0.22909364831922663,0.23717877831500594,0.24524735399056458,0.25750337807458423,0.26567909331061107,0.2775411245733786,0.2932238929430787,0.30753000404322717,0.3211671881718985,0.3345760551160862,0.3515749581606111,0.3713131376806921,0.39457076567968924,0.41639968962636104,0.4456604438756233,0.4815845106655512,0.5201781933515466,0.5638480927900108,0.614196402309182,0.666728557751865,0.7243745413625583,0.7867336061573595,0.8610693905454172,0.9405954178436237,1.0,0.9985286520187862,0.9157860540844209,0.6624627080828389,0.3026165493671681)
V11L12_038_B1_RNA_pgb <- c(0.0,0.007760926938424316,0.012439077881358226,0.014105807151345615,0.015543933327598061,0.01700804324772026,0.01816465642691749,0.019953091937621244,0.020525407456742115,0.02375668122160216,0.027442603188784286,0.028822838122218107,0.02950285827914107,0.030929002314438035,0.0315215124531468,0.03300999054757173,0.033860722555411005,0.033204935941993516,0.03151653111364845,0.03427592393413807,0.03296475459969485,0.0335490522597712,0.03413186898107778,0.033796907557513266,0.03594817306681282,0.03783610073668626,0.03982405908891839,0.040584319201003916,0.042303958374311584,0.04366682593489982,0.044782915244124005,0.04751443948958234,0.05116858858861285,0.05280960337632499,0.05605972545280039,0.06029682590393473,0.06260143589130501,0.06529189774360131,0.06907434999241356,0.07263129565582768,0.07561377170736489,0.07808976669963776,0.08022299168210545,0.08620813840476776,0.09062133593870852,0.09286509280572063,0.0983061964766851,0.10731205439730437,0.11136713401055318,0.11555024751198921,0.12052068191575854,0.125561528226491,0.13114022457224767,0.1358588993851546,0.14044698232472652,0.14762118824872394,0.1539132931891222,0.1589145580454624,0.16591926379571878,0.16843187836484436,0.17529508714718822,0.1815393981545618,0.1852819996873873,0.1913932954670753,0.19873686693401652,0.20438947558746484,0.20900717730243235,0.2174498745981439,0.22392750077715629,0.23197182554379736,0.2399909743513522,0.24640586257884484,0.255304554384851,0.2630357279170825,0.2750165225645928,0.2827051527648925,0.2944024149533888,0.30972663081987056,0.323308454908309,0.3347886922516341,0.34601811241967756,0.36173423853696174,0.38005991339739337,0.40168269648254157,0.4235712407614664,0.45302590121519104,0.48706043212985517,0.5259790991072497,0.5713053842490853,0.6219634528545163,0.6756758903572172,0.7348169108667383,0.7955574798784176,0.8681769280240871,0.9430136107697098,1.0,0.9991685201961679,0.9106117475333282,0.6445023985149685,0.28338234567507314)
V11T17_101_B1_pgb <- c(0.0,0.004676060843130653,0.008669887854104302,0.011410854110879921,0.014473086252290488,0.018189921194212387,0.020135442584962013,0.02147019385675143,0.0226576464649203,0.02445105682079034,0.028645840434922864,0.02898535322724835,0.02993687933444264,0.033430179631381986,0.035458834735631585,0.037589672522948196,0.03938689267273165,0.03932537452680354,0.03925727947348809,0.042122364805653185,0.043395998962452755,0.04467380384101016,0.04575975551870705,0.047259210196837216,0.04942345386131035,0.049873450677124696,0.05136287958179852,0.05332833221017935,0.05447989254937453,0.05598395908329461,0.058153897387090946,0.059225371847916874,0.060551781676190616,0.06124769066334207,0.06363310289026206,0.06621229326654618,0.06857219992579323,0.07224680630991703,0.07732325644180006,0.081210930563418,0.08335171391800329,0.08642172605961597,0.08896077323898091,0.09284860777297414,0.09720709221610177,0.10102627025346599,0.10470625045216238,0.1097334940879217,0.11357332521860597,0.1171837666526092,0.12161848707709884,0.12653171761711265,0.13144230135293403,0.13606586724624728,0.14265576803600552,0.14803030446429244,0.1534172327485714,0.15867254267891553,0.1650361417102173,0.1694418675978712,0.17572754631787027,0.18193418183993956,0.18863925871516424,0.19507524383082075,0.20174667421032597,0.2091815070838907,0.21520575373524226,0.22398881252012173,0.23239546366635638,0.2410943460572964,0.250787865381293,0.2602901330508364,0.27241213532452146,0.28501136441472824,0.2995717551022765,0.31418391888395336,0.3300488632136406,0.3490837966978471,0.3686255530818185,0.3898893363146508,0.41062484149252165,0.4349344145982961,0.4600687078285891,0.489377251888976,0.5193844319468945,0.5503745388344725,0.5849768917952759,0.6241824382777293,0.6707673149518931,0.7201920521122059,0.777974594850474,0.8447740776168111,0.9045997526761997,0.9579171768448587,0.995226889669812,1.0,0.9505297859312893,0.8029888595209477,0.537946349439431,0.23614085201108193)
V11L12_038_C1_pgb <- c(0.0,0.005598045656201522,0.008486606339678964,0.00939845162546907,0.010274661873990006,0.01188274117314016,0.013235907741789035,0.013947965255452723,0.014666583732656946,0.017081339931432786,0.019753517464416583,0.021898952542170756,0.02350092113998414,0.02502692407170567,0.02621831786286004,0.027657420189255495,0.028531121834069757,0.028128909039766323,0.02737645657410798,0.02947268442530816,0.02904905201474004,0.03006805970502551,0.030108904919223925,0.03050860910982069,0.032596281979149384,0.0345236615039388,0.0359808386416567,0.03752382289077726,0.03962532524207865,0.04205050179471298,0.04397022686203844,0.04661416652252916,0.04916638133979635,0.05133574463752188,0.053988496572571945,0.05772014107100783,0.05942901478062872,0.0622457908041921,0.06629139670499405,0.07023833086000017,0.07394990652878265,0.07562398140236994,0.07859320334349275,0.08267405834253208,0.08766090521871261,0.09044519804043316,0.0962106769132622,0.10395872459242747,0.10815883475229178,0.11249370195742488,0.11675877852097485,0.12191859040657198,0.1269868060952895,0.13100919133101174,0.1361503494967387,0.14196973118767528,0.14772060870046802,0.1526746507589459,0.15814366413218361,0.16134194088867737,0.16765564615617268,0.1737969009996271,0.17907918420464752,0.18497067217135796,0.19173586178288266,0.19842277009412473,0.20245371031171847,0.20954559731365852,0.2170489596466651,0.22604944380718048,0.23566395696728337,0.24280389337868202,0.2529513883416952,0.2618156360239866,0.2768426868123411,0.28857330368378165,0.30238271682681955,0.32018962213802893,0.3362383821898914,0.3534187154829799,0.3692981769462278,0.38952106059648034,0.4120196980708902,0.4382245723908093,0.465406644339236,0.49784089624820305,0.5339602226791026,0.5706750530360634,0.6137147808924416,0.6603007738656496,0.713321463992821,0.7725907000142862,0.8349469337047439,0.9046005669315731,0.9669486959023631,1.0,0.9649116453693717,0.8297759295872247,0.5377947199295121,0.22066443649653036)
V11L12_109_B1_RNA_pgb <- c(0.0,0.0034228721971892523,0.0054491535455907375,0.0050305449930453565,0.0065613170670749425,0.00763397679310032,0.00836308517795209,0.007884957573582074,0.008490682437284104,0.011317310681681444,0.014939018649346799,0.0176287004028202,0.018921264590233054,0.020781893356343856,0.021788318385316596,0.023523983478459305,0.024718051535861478,0.023405867130037408,0.021776203888042557,0.024069399214427523,0.023050201291589762,0.02415749267830071,0.02442809367719368,0.024562406581753696,0.027015723959065114,0.02957280463911505,0.031066969862686124,0.03207115634324953,0.034094540746650706,0.03578306464387921,0.03687126252945173,0.03931786426153514,0.042749558973043535,0.04422818603706746,0.046623037797099816,0.05095792095211521,0.053964159786532025,0.05711287564323701,0.062015164980359375,0.06599767427988203,0.06984652907143564,0.07176377994437076,0.07401246766120544,0.07891423028105501,0.08331455805721387,0.08496752853852943,0.09125337247193815,0.10005336962766488,0.10430384633971658,0.10876514164020022,0.11438956035830995,0.12018977096621149,0.12650484770825973,0.1314748202149349,0.1374707062896758,0.14455123490839794,0.15130612057322115,0.15635444227422296,0.16261908583740212,0.1658452291332064,0.17316541411104544,0.17868830807482597,0.18316224458985644,0.1885205394058919,0.19415536079013912,0.2001461113714704,0.20325927381226241,0.2113838877449614,0.217774416736336,0.22617305533020104,0.2345161252517873,0.24069307038904786,0.248583426814676,0.25629535776414836,0.2689212975100626,0.27765071455123624,0.28964367181458184,0.3057564799063286,0.32021710759266897,0.333208194087941,0.34643459152482936,0.3647738335286357,0.3853622899665495,0.41043337375449473,0.4330167718630793,0.46401052802484843,0.5014836967177745,0.5408059063968101,0.5855292066702948,0.6343084933028558,0.6845683821892135,0.7417835726626888,0.8041794225519531,0.8790017180200645,0.9553707187597072,1.0,0.9797011063827994,0.8925353233063044,0.6483465620571048,0.30078900930669916)
V11L12_038_D1_RNA_pgb <- c(0.0,0.003923213758017881,0.006570635578243512,0.007587421704339362,0.00862130105514448,0.010473221580341053,0.011697907673409317,0.012785616196264985,0.013565009299873105,0.016050422115735617,0.019415121999269917,0.02164344112687808,0.023406303054182623,0.025943053487307556,0.027498131333908903,0.02915605824443891,0.030394940405486057,0.03023837805577606,0.02983196490963768,0.031706715029869686,0.03185394853490784,0.03235063708476501,0.033232589536628755,0.0338292299936842,0.036003951721780246,0.038076925305505174,0.039868353198171315,0.041834595527948874,0.04404246072904052,0.046478448051082664,0.04896924957846369,0.0523693527172433,0.055011385825950415,0.05722290144452235,0.06034470370778118,0.06410741498293575,0.06594751510867235,0.06879959208033236,0.0735160473511296,0.07760399344083717,0.08142661791717608,0.08344222779763939,0.08688387616393271,0.09160351830713338,0.09679771472277107,0.10011397414577333,0.10578695468267442,0.1127383925415597,0.11693556144000278,0.12170277489671637,0.1266488588099639,0.132470869089076,0.13823487829044576,0.14278996193136057,0.14854504789000075,0.15484219187289594,0.16088392251844041,0.1659071866869854,0.17165989697710668,0.17552412462409392,0.18308715226876343,0.18977518063772214,0.1963417022534085,0.20220160734255402,0.20886414073228535,0.21574257024156493,0.2202627141723113,0.22815503265095635,0.2366435859847146,0.2462265113018084,0.25659352311641354,0.2643497331718651,0.2752870213172792,0.28548605598465665,0.30179183349460836,0.3154928932745404,0.33130111308761584,0.350128054327483,0.3672427759396928,0.38641812924795604,0.40471952625693147,0.42667545470874885,0.4503099957701512,0.47856695039488245,0.5066193657544485,0.5391502059878436,0.5742665268305684,0.6113993267007759,0.6552157512617117,0.7006654189578347,0.7514599931627102,0.8067425528586245,0.8655384365783421,0.9306048104390351,0.9830047571313513,1.0,0.946248587636094,0.7925164123928776,0.5081046800669823,0.2155729706865682)
V11T17_101_A1_pgb <- c(0.0,0.0036270043524800582,0.006678257243999471,0.00929835134950936,0.011565703472763257,0.01408851156407313,0.016139375578805012,0.018191736301447665,0.01991305730654048,0.02262439714075335,0.026755685151454812,0.028437396850766194,0.030224091919247263,0.03349375725803111,0.03562517623067476,0.0379087783255313,0.04010738879259771,0.03996006997109483,0.040897810931045865,0.04377811839761382,0.04561121141132874,0.046639984284566935,0.04784307017555582,0.049692787337851076,0.052808559031095444,0.05467832138862422,0.05640306344037022,0.05914952245663222,0.06180986731417102,0.06388430447849752,0.0669004381879678,0.06944607107491692,0.07177334496845554,0.07265554737414624,0.07592569379761568,0.07923716005019317,0.08155946273531389,0.08463289897622506,0.08952312480539457,0.09360705270149096,0.09624590910973943,0.09947815675075403,0.10208290960017853,0.10617266396635615,0.11114173423011083,0.11514569516068897,0.11948823280222474,0.12518801724955866,0.12916813776127242,0.13318850902501506,0.13793617342568376,0.1436850819648033,0.14923156075899127,0.15363979318703905,0.1601150861475674,0.16536821019124454,0.1708583486233628,0.17667573154940006,0.18368673903427596,0.1879930884166835,0.1944033280369473,0.19966869301318038,0.20581005974804525,0.21125107408837793,0.21753981309859965,0.2244523585844885,0.22935781876203548,0.23821848995535266,0.24625436818212798,0.25473642572787447,0.2640897810931046,0.27354175881888365,0.28520651228302746,0.2959890098876266,0.3101443387691448,0.32316446815419836,0.33755889612446194,0.3541278268736578,0.3714830638145472,0.39155386344411197,0.4118869076811852,0.4359571781175958,0.4599077119211449,0.48750818178052113,0.5162546488148613,0.545967294259457,0.5790056246817825,0.6163560775027429,0.6590419198486187,0.7040821371919889,0.7557737244908186,0.8183478750088533,0.8814229950461641,0.9390146049292472,0.9828535538126629,1.0,0.9692431869722267,0.8494725040056982,0.6049789057728826,0.2895191692202018)
rseqc_output <- data.frame(V11L12_109_D1_pgb,V11L12_109_C1_RNA_pgb,V11L12_038_A1_RNA_pgb,V11L12_109_A1_RNA_pgb,V11L12_038_B1_RNA_pgb,V11T17_101_B1_pgb,V11L12_038_C1_pgb,V11L12_109_B1_RNA_pgb,V11L12_038_D1_RNA_pgb,V11T17_101_A1_pgb)
write.csv(rseqc_output, file = paste0(tables.dir, "/rseqc_output.csv"))

```

# 3 Spatial analysis striatum
For this part of the analysis we will work on an object that contains only 10x filtered bc matrices of Striatum samples, but with the additional filters applied with STUtility.
<!-- Upload object -->
```{r message=F, warning=F }

se.all <- readRDS(paste0(project.dir, "/R_objects/allRNA_10xFFBM_STUF"))
se_mPDStr <- SubsetSTData(se.all, expression = Condition == "mPD" & Brain.area == "Striatum" & Data.Type == "RNA")
# keep only barcodes with spatial annotation and remove barcodes under artifacts
se_mPDStr$RegionLoupe[is.na(se_mPDStr$RegionLoupe)] <- "out"
se <- SubsetSTData(se_mPDStr, expression = RegionLoupe != "out")

```

<!-- Adjust H&E images -->
```{r message=F, warning=F}

se <- se %>% MaskImages(verbose = FALSE, channels.use = 3) %>% 
  WarpImages(transforms = list("1" = list(angle = 140), "2" = list(angle = -45, "mirror.x" = T), 
                                        "3" = list(angle = 135), "4" = list(angle = -135, "mirror.x" = T), 
                                        "5" = list(angle = 140, "mirror.x" = T), 
                                        "6" = list(angle = -40), "7" = list(angle = -28), 
                                        "8" = list(angle = -40, "mirror.x" = T), 
                                        "9" = list(angle = -40))) 

```

<!-- Check H&E images -->
```{r}

ImagePlot(se, annotate = FALSE, type = "processed")

```

<!-- Check that lesion side is always on the right -->
```{r message=F, warning=F}

ST.FeaturePlot(se, features="lesion", ncol=3, label.by = "Sample.ID", cols = my_cols)

```

<!-- Standard spatial transcriptomics analysis -->
```{r message=F, warning=F}

se <- SCTransform(se, verbose=F) 
se <- RunPCA(se, dims.use = 1:30, verbose=F) 

```

<!-- Harmony data integration -->
```{r message=F, warning=F}

se <- RunHarmony(se, group.by.vars = c("Sample.ID"), assay.use = "SCT", reduction = "pca", dims.use = 1:30, verbose = F) 
se <- FindNeighbors(se, reduction = "harmony", dims = 1:30, verbose = F)
se <- RunUMAP(se, reduction = "harmony", dims = 1:30, reduction.name = "umap.harmony", verbose=F)
se <- FindClusters(se, verbose = T, resolution = 0.36)

```

<!-- Differential gene expression analysis. Find all markers per RNA cluster -->
```{r message=F, warning=F}

CL.mPDStr <- FindAllMarkers(se, only.pos = T, 
                 logfc.threshold = 0.25, min.pct = 0, min.diff.pct = 0, min.cells.group = 0, min.cells.feature = 0)
write.csv(CL.mPDStr, file = paste0(tables.dir, "/DGE_mPDStr_CL_logfc025_noThresholds.csv"))

```

## Adding replicate column
```{r}

se$label <- paste(se$Laser, se$Matrix, se$Prot.Short, se$sample, se$section_number, sep = ".")
se$letters <- LETTERS[se$seurat_clusters]
se$clusters_type <- paste(se$letters, se$label, sep = "_")

# add replicate column
se$MatSn <-  paste0(se$Matrix, se$section_number)
df <- se@meta.data %>% 
  group_by(Matrix) %>%
  mutate(repl = as.numeric(as.factor(section_number)))
df$repl <- paste0("rep", df$repl)
se$repl <- df$repl
table(se$label, se$repl)

```


## Stereoscope.
Previous analysis showed markers of MSN2 neurons differentially expressed in the lesioned hemisphere, in a cluster located in the lesioned caudate and enriched in a factor that localizes in the lesioned caudate. Following are the results of the cell type deconvolution analysis.
In the dataset that we used (see 4 Substantia Nigra analysis) there are 6 types of medium spiny neurons annotated: MSN1 and MSN2 are expected to localize in the dorsal striatum, MSN3-6 in the ventral. Stereoscope results confirm expected localization.
MSN1 neurons are D1 type MSNs. Indeed their markers are Drd1, Ido1 and Ppp1r1b, two of which we found in the factor that localizes to the intact caudate.

<!-- Upload stereoscope results -->
```{r}

m_stsc <- read.csv(file=paste0(data.dir, "/misc/Zeisel_stsc_output.csv"), header = T, row.names = 1)
se <- AddMetaData(se, metadata = m_stsc)

```

<!-- Save Seurat object -->
Saving Seurat object with calculations. The acronyms in the name mean:
- mPDStrRNA: contains only mouse Striatum sections
- 10xFFBM: contains 10x filtered feature barcode matrix
- STUF: additional filters with STUtility were applied
- An: contains additional calculations (mainly run with Seurat)
```{r message=F, warning=F}

saveRDS(se, file = paste0(project.dir, "/R_objects/mPDStrRNA.10xFFBM.STUF.An"))

```




# 4 Substantia nigra

We will start uploading the object that contains the whole dataset, and subsetting only for the conditions of interest. This object already contains stereoscope results in the metadata, that will be used later on.

Uplaod Seurat objects
```{r message=F, warning=F}

se.all <- readRDS(paste0(project.dir, "/R_objects/allRNA_10xFFBM_STUF"))
se_mPDSubNig <- SubsetSTData(se.all, expression = Condition == "mPD" & Brain.area == "SubNigra" & Data.Type == "RNA")
# keep only barcodes with spatial annotation and remove barcodes under artifacts
se_mPDSubNig$RegionLoupe[is.na(se_mPDSubNig$RegionLoupe)] <- "out"
se <- SubsetSTData(se_mPDSubNig, expression = RegionLoupe != "out")

```

Here we will adjust the images. We will put all of them horizontally, with lesioned side always on the right, to facilitate understanding of the results. 
```{r fig.width=4*3, fig.height=4*2}

ImagePlot(se, annotate = FALSE, type = "raw")
se <- se %>% MaskImages(verbose = FALSE, channels.use = 3) %>% 
  WarpImages(transforms = list("1" = list(angle = 140, "mirror.x" = T),"2" = list(angle = -45), 
                               "3" = list(angle = -50), "4" = list(angle = -45), "5" = list(angle = -40)
                               ))
ImagePlot(se, annotate = FALSE, type = "processed")

```

Let's check that lesioned side is always on the right.
```{r message=F, warning=F, fig.width=4*3, fig.height=4*2}

ST.FeaturePlot(se,
               features="lesion",
               ncol=3)

```

Spatial Transcriptomics data analysis
```{r message=F, warning=F}

se <- SCTransform(se, verbose=F) 
se <- RunPCA(se, verbose=F) 
se <- RunUMAP(se, reduction = "pca", dims = 1:30, verbose=F) 
se <- FindNeighbors(se, reduction = "pca", dims = 1:30, verbose=F) 

```

We will integrate the data with harmony to account for batch effects between sections
```{r message=F, warning=F}

se <- RunHarmony(se, group.by.vars = c("Sample.ID"), assay.use = "SCT", reduction = "pca", dims.use = 1:30, verbose = F) %>%
  FindNeighbors(reduction = "harmony", dims = 1:30, verbose = F) %>%
  RunUMAP(reduction = "harmony", dims = 1:30, reduction.name = "umap.harmony", verbose=F)

```

<!-- Upload stereoscope results -->
```{r}

m_stsc <- read.csv(file=paste0(data.dir, "/misc/Zeisel_stsc_output.csv"), header = T, row.names = 1)
se <- AddMetaData(se, metadata = m_stsc)

```

Save Seurat object.
```{r message=F, warning=F}

saveRDS(se, file = paste0(project.dir, "/R_objects/mPDSubNig.10xFFBM.STUF.An"))

```

# 5 human Striatum

Here we will load only the human sample processed with RNArescue because is the only one that we have dopamine profile for. There is also another human sample in the dataset  which was analyzed with Visium (no MSI). It's in the dataset just for QC purposes, will not be part of the analysis.
```{r message=F, warning=F}

se.all <- readRDS(paste0(project.dir, "/R_objects/allRNA_10xFFBM_STUF"))
se <- SubsetSTData(se.all, expression = Condition == "hPD" & Data.Type == "RNA" & Visium.Protocol == "RNArescue" )

```

These are the H&E images of the sections that we are working on.
```{r message=FALSE, warning=FALSE}

se <- se %>% MaskImages(verbose = FALSE, channels.use = 2)

```

```{r, warning=FALSE, echo=FALSE}

se <- se %>% 
  SCTransform(verbose=F) %>%
  RunPCA(verbose = F) %>%
  RunHarmony(assay.use = "SCT", reduction = "pca", group.by.vars = "Sample.ID",
                     reduction.save = "harmony", verbose = F) %>%
  FindNeighbors(reduction = "harmony", dims = 1:30, verbose = F) %>%
  RunUMAP(reduction = "harmony", dims = 1:30, verbose = F)

```

Adding stereoscope results
```{r}

h_stsc <- read.csv(file=paste0(data.dir, "/misc/Kamath_stsc_output.csv"), header = T, row.names = 1)
#changing names of the clusters
h_stsc <- h_stsc %>% rename("Astro_0" = "X0",
                                          "Oligo_1" = "X1",
                                          "MSN.D2.PenkPos_2" = "X2",
                                          "MSN.D1.TacNeg_3" = "X3",
                                          "Oligo_4" = "X4",
                                          "MSN.D2.PenkNeg_5" = "X5",
                                          "MSN.D1.TacPosB_6" = "X6",
                                          "OPC_7" = "X7",
                                          "Mi.MiR.Ma_8" = "X8",
                                          "MSN.D1.TacPosA_9" = "X9",
                                          "Astro_10" = "X10",
                                          "Astro_11" = "X11",
                                          "Inhib_MSN_12" = "X12",
                                          "Inhib_13" = "X13",
                                          "unk_14" = "X14",
                                          "unk_15" = "X15",
                                          "unk_16" = "X16",
                                          "Astro_17" = "X17",
                                          "MSN.D2_C_18" = "X18",
                                          "unk_19" = "X19",
                                          "OPC_20" = "X20",
                                          "Oligo_21" = "X21",
                                          "unk_22" = "X22",
                                        )

# for some reason it adds an X at the beginning of each cluster
se <- AddMetaData(se, metadata = h_stsc)

```

Save Seurat object.
```{r message=F, warning=F}

saveRDS(se, file = paste0(project.dir, "/R_objects/hPDStr.10xFFBM.STUnF.An"))

```


# Z multimodal analysis
Multimodal Analysis: Dopamine-Gene correlations
Here we will the /data/misc/se.multi.list object that contains both modalities aligned and integrated. We will leverage the fact that the 2 modalities are integrated to calculate dopamine-gene correlations spot by spot. The correlations are calculated pairwise: each gene vs dopamine. In the end in the main.Rmd file we will make a volcano plot of the results. In case you want to reproduce the se.multi.list object from scratch you can follow the MSI_SRT_mPD_LL.Rmd and MSI_SRT_hPDStr_LLMV.Rmd scripts. Otherwise you can use the one that we provide in the /data/misc folder to complete this script and reproduce the figures of the article.

## Striatum
<!-- You can upload the object with all the analysis done -->
```{r message=F, warning=F}

se <- readRDS(file = paste0(project.dir, "/R_objects/mPDStrRNA.10xFFBM.STUF.An"))
se_mPDStrFMP10 <- SubsetSTData(se, expression = Protocol == "Sma" & Matrix == "FMP-10")
md <- se_mPDStrFMP10@meta.data
se.multi.list <- readRDS(paste0(data.dir, "/misc/se.multi.list"))
se.1 <- se.multi.list$`V11L12-109_A1`
mm <- MergeSTData(se.1, list(se.multi.list$`V11L12-109_B1`, se.multi.list$`V11L12-109_C1`))

rownames(md) <- gsub("_5", "_1_1"  ,rownames(md))
rownames(md) <- gsub("_6", "_1_2"  ,rownames(md))
rownames(md) <- gsub("_7", "_1_3"  ,rownames(md))

mm <- AddMetaData(mm, md)

```

<!-- Dopamine genes correlation analysis -->
```{r message=F, warning=F}

Idents(mm) <- mm$RegionLoupe
seCPACB <- SubsetSTData(mm, idents = c("CP", "ACB")) 
dop_vals <- GetAssayData(seCPACB, slot = "data", assay = "MSI")["674.2805", ]
rna <- GetAssayData(seCPACB, slot = "data", assay = "RNA")
experiment <- t(as.matrix(rna))

cormstr = lapply(colnames(experiment), function(g){
  cor_out = cor.test(x = dop_vals,
                     y = experiment[,g], 
                     method = 'pearson')
  return(
    data.frame(
      dopamine = 'Dopamine',
      gene = g,
      rho = cor_out$estimate,
      Pvalue = cor_out$p.value
    )
  )
}) %>% data.table::rbindlist()

cormstr$FDR = p.adjust(cormstr$Pvalue, method = 'BH') ## alternative, method='bonferroni'
cormstr$sig <- "NO"
cormstr$sig[cormstr$rho > 0.2 & cormstr$FDR < 0.01] <- "PosCor"
cormstr$sig[cormstr$rho < -0.2 & cormstr$FDR < 0.01] <- "NegCor"
mycolors <- c("blue", "red", "grey")
names(mycolors) <- c("NegCor", "PosCor", "NO")
cormstr$corlabel <- NA
cormstr$corlabel[cormstr$sig != "NO"] <- cormstr$gene[cormstr$sig != "NO"]
write.csv(cormstr, file = paste0(tables.dir, "/cormstr_onlyCPACBSpots.csv"))

```

## Substantia nigra
<!-- Upload list with multimodal objects -->
```{r message=F, warning=F}

se <- readRDS(file = paste0(project.dir, "/R_objects/mPDSubNig.10xFFBM.STUF.An"))
# Upload multimodal object and select only substantia nigra samples
se.multi.list <- readRDS(paste0(data.dir, "/misc/se.multi.list"))
se.1 <- se.multi.list$`V11T16-085_A1`
mm <- MergeSTData(se.1, list(se.multi.list$`V11T16-085_B1`, se.multi.list$`V11T16-085_C1`))

# Fetch metadata from the object that we are using and modify rownames to make them as the multimodal object
md <- se@meta.data
rownames(md) <- gsub("_9", "_1_1"  ,rownames(md))
rownames(md) <- gsub("_10", "_1_2"  ,rownames(md))
rownames(md) <- gsub("_11", "_1_3"  ,rownames(md))

#Add metadata of the object we are using for the whole analysis to the multimodal
mm <- AddMetaData(mm, md)

```

<!-- Dopamine genes correlation analysis -->
```{r message=F, warning=F}

Idents(mm) <- mm$RegionLoupe
# There was a typo in the annotation. It should be gone but check it before removing the following line
se$RegionLoupe[se$RegionLoupe == "VTA-SNpc"] <- "VTA-SNc"
mmSN <- SubsetSTData(mm, idents = "VTA-SNc") 
dop_vals <- GetAssayData(mmSN, slot = "data", assay = "MSI")["674.2805", ]
rna <- GetAssayData(mmSN, slot = "data", assay = "RNA")
experiment <- t(as.matrix(rna))

cormsubnig = lapply(colnames(experiment), function(g){
  cor_out = cor.test(x = dop_vals,
                     y = experiment[,g], 
                     method = 'pearson')
  return(
    data.frame(
      dopamine = 'Dopamine',
      gene = g,
      rho = cor_out$estimate,
      Pvalue = cor_out$p.value
    )
  )
}) %>% data.table::rbindlist()

cormsubnig$FDR = p.adjust(cormsubnig$Pvalue, method = 'BH') ## alternative, method='bonferroni'
cormsubnig$sig <- "NO"
cormsubnig$sig[cormsubnig$rho > 0.3 & cormsubnig$FDR < 0.01] <- "PosCor"
cormsubnig$sig[cormsubnig$rho < -0.3 & cormsubnig$FDR < 0.01] <- "NegCor"
cormsubnig$corlabel <- NA
cormsubnig$corlabel[cormsubnig$sig != "NO"] <- cormsubnig$gene[cormsubnig$sig != "NO"]
cormsubnig$InvLog10FDR <- -log10(cormsubnig$FDR)
cormsubnig$InvLog10FDR[is.infinite(cormsubnig$InvLog10FDR)] <- max(na.omit(cormsubnig$InvLog10FDR[!is.infinite(cormsubnig$InvLog10FDR)]))+50
write.csv(cormsubnig, file = paste0(tables.dir, "/cormsubnig_onlySubNigSpots.csv"))

```

## Human Striatum
<!-- You can upload the object with all the analysis done -->
```{r message=F, warning=F}

se <- readRDS(file = paste0(project.dir, "/R_objects/hPDStr.10xFFBM.STUnF.An"))
md <- se@meta.data

hPDStr.multi.list <- readRDS(paste0(data.dir, "/misc/hPDStr.multi.list"))
hPDCdMM <- MergeSTData(hPDStr.multi.list[[1]], 
                        y= list(hPDStr.multi.list[[2]], hPDStr.multi.list[[3]]))

rownames(md) <- gsub("_13", "_1_1"  ,rownames(md))
rownames(md) <- gsub("_14", "_1_2"  ,rownames(md))
rownames(md) <- gsub("_15", "_1_3"  ,rownames(md))

hPDCdMM <- AddMetaData(hPDCdMM, md)

```

Dopamine genes correlation
```{r message=FALSE, warning=FALSE, echo=FALSE}

Idents(hPDCdMM) <- hPDCdMM$RegionLoupe
mm <- SubsetSTData(hPDCdMM, idents = "Cd") 

DefaultAssay(mm) <- "MSI"
mm <- NormalizeData(mm)
mm <- ScaleData(mm)
dop_vals <- GetAssayData(mm, slot = "data", assay = "MSI")["674.2805", ]
DefaultAssay(mm) <- "RNA"
mm <- NormalizeData(mm)
mm <- ScaleData(mm)
rna <- GetAssayData(mm, slot = "data", assay = "RNA")
experiment <- t(as.matrix(rna))
dim(experiment)[1] == length(dop_vals)

corHstr = lapply(colnames(experiment), function(g){
  cor_out = cor.test(x = dop_vals,
                     y = experiment[,g], 
                     method = "pearson")
  return(
    data.frame(
      dopamine = 'Dopamine',
      gene = g,
      rho = cor_out$estimate,
      Pvalue = cor_out$p.value
    )
  )
}) %>% data.table::rbindlist()

corHstr$FDR = p.adjust(corHstr$Pvalue, method = 'BH') ## alternative, method='bonferroni'
corHstr$sig <- "NO"
corHstr$sig[corHstr$rho > 0.2 & corHstr$FDR < 0.01] <- "PosCor"
corHstr$sig[corHstr$rho < -0.2 & corHstr$FDR < 0.01] <- "NegCor"
mycolors <- c("blue", "red", "grey")
names(mycolors) <- c("NegCor", "PosCor", "NO")
corHstr$corlabel <- NA
corHstr$corlabel[corHstr$sig != "NO"] <- corHstr$gene[corHstr$sig != "NO"]
corHstr$InvLog10FDR <- -log10(corHstr$FDR)
corHstr$InvLog10FDR[is.infinite(corHstr$InvLog10FDR)] <- max(na.omit(corHstr$InvLog10FDR[!is.infinite(corHstr$InvLog10FDR)]))+50
write.csv(corHstr, file = paste0(tables.dir, "/corHstr_onlyCdSpots.csv"))

```
